pages/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { request, gql } from 'graphql-request';
+import Link from 'next/link';
+
+export type BlogPostPreview = {
```

Comment:
Usually we add typescript types and interfaces and stuff to an `index.d.ts` file at root level, and that way they're available across the whole project, and it keeps things cleaner.
Another approach is to add them to a file like `types.ts` and export them from there, though in this case `index.d.ts` should be enough.